timelock-agent/src/main/java/com/palantir/timelock/paxos/AsyncTimeLockServicesCreator.java

```diff
@@ -82,16 +83,16 @@ public TimeLockServices createTimeLockServices(
     private static AsyncTimelockService createRawAsyncTimelockService(
             String client,
             Supplier<ManagedTimestampService> timestampServiceSupplier) {
-        ScheduledExecutorService reaperExecutor = Executors.newSingleThreadScheduledExecutor(
-                new ThreadFactoryBuilder()
+        ScheduledExecutorService reaperExecutor = new InstrumentedScheduledExecutorService(
+                Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
                         .setNameFormat("async-lock-reaper-" + client + "-%d")
                         .setDaemon(true)
-                        .build());
-        ScheduledExecutorService timeoutExecutor = Executors.newSingleThreadScheduledExecutor(
-                new ThreadFactoryBuilder()
+                        .build()), AtlasDbMetrics.getMetricRegistry(), "async-lock-reaper");
+        ScheduledExecutorService timeoutExecutor = new InstrumentedScheduledExecutorService(
+                Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
                         .setNameFormat("async-lock-timeouts-" + client + "-%d")
                         .setDaemon(true)
-                        .build());
+                        .build()), AtlasDbMetrics.getMetricRegistry(), "async-lock-timeouts");
```

Comment:
this static name will be passed to the executor for each client - does `InstrumentedScheduledExecutorService` correctly aggregate across executors of the same name?

Reply:
Yup, this is similar to registering metics like meter for `async-lock-timeouts-executor.running_tasks`, meter for `async-lock-timeouts-executor.blocked_tasks` and two more.